### PR TITLE
Add example test code

### LINKED ISSUE #13: Add example test code
Add example test code into this project so that users can see how to test gRPC services in a Play Java application. Depends on: https://github.com/akka/akka-grpc/issues/361

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes #13

### CODE REVIEW

File: test/test/HelloFunctionalTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    return wsClient.url(url).get().toCompletableFuture().get();
+  }
+
+  @Test public void returns404OnNonGrpcRequest() throws Exception {
+    assertEquals(404, wsGet("/").getStatus()); // Maybe should be a 426, see #396
+  }
+
+  @Test public void returns200OnNonExistentGrpcMethod() throws Exception {
+    final WSResponse rsp = wsGet("/" + GreeterService.name + "/FooBar");
+    assertEquals(200, rsp.getStatus()); // Maybe should be a 426, see #396
```
Comment (raboof) [inline test/test/HelloFunctionalTest.java]: Kind of weird to have a test when we're not yet decided about what the desirable behavior is, but OK.

File: test/test/HelloFunctionalTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  @Test public void returns200OnEmptyRequestToAGrpcMethod() throws Exception {
+    final WSResponse rsp = wsGet("/" + GreeterService.name + "/SayHello");
+    assertEquals(200, rsp.getStatus()); // Maybe should be a 426, see #396
+  }
+
+  @Test public void worksWithAGrpcClient() throws Exception {
```
Comment (raboof) [inline test/test/HelloFunctionalTest.java]: This is the most interesting one. Perhaps creating and closing the `greeterServiceClient` could be moved out of this test to make it a bit easier on the eyes?

File: test/test/HelloFunctionalTest.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  }
+
+  @Test public void returns200OnEmptyRequestToAGrpcMethod() throws Exception {
+    final WSResponse rsp = wsGet("/" + GreeterService.name + "/SayHello");
+    assertEquals(200, rsp.getStatus());
+  }
+
+  @Test public void worksWithAGrpcClient() throws Exception {
+    final HelloRequest req = HelloRequest.newBuilder().setName("Alice").build();
+    try (final CloseableGreeterServiceClient greeterServiceClient = newGreeterServiceClient()) {
```
Comment (marcospereira) [inline test/test/HelloFunctionalTest.java]: I like try-with-resources, but here it looks like using a try-finally would be simpler than having a wrapper.

```java
@Test public void worksWithAGrpcClient() throws Exception {
  GreaterServiceClient greeterServiceClient = newGreeterServiceClient();
  final HelloRequest req = HelloRequest.newBuilder().setName("Alice").build();

  try {
    final HelloReply helloReply = greeterServiceClient.sayHello(req).toCompletableFuture().get();
    assertEquals("Hello Alice!", helloReply.getMessage());
  } finally {
    greeterServiceClient.close().toCompletableFuture().get(30, TimeUnit.SECONDS);
  }
}
```

File: test/test/HelloFunctionalTest.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      final HelloReply helloReply = greeterServiceClient._0.sayHello(req).toCompletableFuture().get();
```
Comment (marcospereira) [inline test/test/HelloFunctionalTest.java]: Use `get` with a timeout?

**Comment by raboof** [comment]
Looks like the runtime was cross-published, but the play-testkit wasn't yet

**Comment by dwijnand** [comment]
(╯°□°）╯︵ ┻━┻

**Pull request merged**